### PR TITLE
Add graph-union!

### PR DESCRIPTION
**Pull request by LeifAndersen**

### CODE REVIEW

**Comment by stchang** [comment]
No tests?

**Comment by LeifAndersen** [comment]
@stchang FWIW, I've added tests.

File: graph/tests/boost-examples.rkt [hunk 1]
```diff
@@ -135,6 +135,14 @@
 (check-false (eq? g/for-copy g/copy))
 (check-equal? g/for-copy g/copy)
 
+;; graph union
+(define g/for-union
+  (undirected-graph '((a c) (a d) (b a) (b d))))
+(define g/for-union2
+  (undirected-graph '((c f) (d c) (d e) (d f) (e b) (e g) (f e) (f g))))
+(graph-union! g/for-union g/for-union2)
+(check-equal? g/for-union g/for-union)
```
Comment (stchang) [inline graph/tests/boost-examples.rkt]: Sorry to nit, but won't `check-equal`ing the same two things always pass?  I don't see how this tests the `graph-union` operation.

Comment (LeifAndersen) [inline graph/tests/boost-examples.rkt]: Eep...you are correct. I actually meant to test it against the graph above. Good catch.

Comment (LeifAndersen) [inline graph/tests/boost-examples.rkt]: And fixed, thanks.

**Comment by stchang** [comment]
Thanks Leif

**Pull request merged**